cstar/case.py

```diff
@@ -381,8 +381,8 @@ def from_blueprint(
                     model_grid = [
                         ROMSModelGrid(
                             base_model=base_model,
-                            source=DataSource(f["source"]),
-                            file_hash=f["hash"],
+                            source=DataSource(f.get("source")),
+                            file_hash=f.get("hash", None),
```

Comment:
What is the type of `f` here? Couldn't we just change `f.get` to return `None` if it doesn't have a hash instead of adding this `, None`?

Reply:
This is a list comprehension that ruff made a bit screwy looking:
`for f in input_dataset_info["model_grid"]["files"]` is at the end. `f` is a dictionary in this case from parsing this section of the blueprint yaml file:
```
      input_datasets:
        model_grid:
          files:
            - location: 'https://github.com/CWorthy-ocean/input_datasets_roms_marbl_example/raw/main/roms_grd.nc'
              hash: 'fd537ef8159fabb18e38495ec8d44e2fa1b7fb615fcb1417dd4c0e1bb5f4e41d'

```
`f.get` here is a python builtin for cleaner dictionary access, the `,None` part is doing exactly what you say: no error if the key isn't present, `get` just returns `None`.

Reply:
Okay, I wasn't sure if the type we were calling `.get` on was a `DataSource` or something. But if it's a `dict` this seems fine.